crates/oxc_linter/src/rules/react_perf/no_jsx_as_prop.rs

```diff
@@ -0,0 +1,94 @@
+use oxc_ast::{
+    ast::{Expression, JSXAttributeValue, JSXElement, JSXExpression, JSXExpressionContainer},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, utils::get_prop_value, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error(
+    "eslint-plugin-react-perf(no-jsx-as-prop): JSX attribute values should not contain other JSX."
+)]
+#[diagnostic(severity(warning), help(r"simplify props or memoize props in the parent component (https://react.dev/reference/react/memo#my-component-rerenders-when-a-prop-is-an-object-or-array)."))]
+struct NoJsxAsPropDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoJsxAsProp;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Prevent JSX that are local to the current method from being used as values of JSX props
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // Bad
+    /// <Item jsx={<SubItem />} />
+    /// <Item jsx={this.props.jsx || <SubItem />} />
+    /// <Item jsx={this.props.jsx ? this.props.jsx : <SubItem />} />
+    ///
+    /// // Good
+    /// <Item callback={this.props.jsx} />
+    /// ```
+    NoJsxAsProp,
+    correctness
```

Comment:
are these `correctness`? or should it be `restriction`?

i don't think there is anything inherintly wrong with the code

Reply:
I wonder if `pedantic` would be a more appropriate category.